src/Runner.Worker/ExecutionContext.cs

```diff
@@ -70,6 +69,10 @@ public interface IExecutionContext : IRunnerService
 
         bool EchoOnActionCommand { get; set; }
 
+        IExecutionContext FinalizeContext { get; set; }
```

Comment:
we'll be able to clean this up when the handler performs the steps-runner-ish things